### PR TITLE
feat(provisioner): default elastic search resource provisioner

### PR DESCRIPTION
**Pull request by giesan**
This pull request is a proposal for solving the issue #127 

#### Description
This pull request implements a provisioner that prepares a service with Elasticsearch (image `docker.elastic.co/elasticsearch/elasticsearch:8.14.0`) with the corresponding parameters and certificates.

This provisions a single elasticsearch container with persistent data and healthcheck.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New chore (expected functionality to be implemented)

#### Checklist:

- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] I've signed off with an email address that matches the commit author.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,8 +67,9 @@ See the [examples](./examples) for more examples of using Score and provisioning
 - [09-dns-and-route](examples/09-dns-and-route) - an example of using the `dns` and `route` resources to route http requests
 - [10-amqp-rabbitmq](examples/10-amqp-rabbitmq) - an example the default `amqp` resource provisioner
 - [11-mongodb-document-database](examples/11-mongodb-document-database) - an example the default `mongodb` resource provisioner
-- [12-mysql-database](examples/12-mysql-database) - an example of the `mysql` resource provisioner
+- [12-mysql-database](examples/12-mysql-database) - an example the default `mysql` resource provisioner
```
Comment (astromechza) [inline README.md:70]: typo here?
```suggestion
- [12-mysql-database](examples/12-mysql-database) - an example of the default `mysql` resource provisioner
```

File: README.md [hunk 2]
```diff
@@ -66,8 +67,9 @@ See the [examples](./examples) for more examples of using Score and provisioning
...
 - [13-kafka-topic](examples/13-kafka-topic) - an example of the default `kafka-topic` resource provisioner
+- [14-elasticsearch](examples/14-elasticsearch) - an example the default `elasticsearch` resource provisioner
```
Comment (astromechza) [inline README.md:72]: typo
```suggestion
- [14-elasticsearch](examples/14-elasticsearch) - an example of the default `elasticsearch` resource provisioner
```

File: examples/14-elasticsearch/score.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  name: example
+
+containers:
+  hello:
+    image: busybox
+    command: ["/bin/sh", "-c"]
+    args:
+      [
+        "echo",
+        "curl --cacert /tmp/ca.crt -u $ECS_USER:$ECS_PASS https://$ECS_HOST:$ECS_PORT",
```
Comment (astromechza) [inline examples/14-elasticsearch/score.yaml:13]: In this example, wehere does the ca.crt come from? There are no volume mounts? Does it require a manual step?

File: README.md (hunk 1 above)
Comment (giesan) [reply README.md:70]: Yes, this is a typo. Thanks.

File: README.md (hunk 2 above)
Comment (giesan) [reply README.md:72]: Yes, this is a typo. Thanks.

File: examples/14-elasticsearch/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+```yaml
+resources:
+  ecs:
+    type: elasticsearch
+    # Optionally, you can overwrite the port that will be published, elasticsearch stack version,
+    # and elasticsearch memory limit by commenting out the following lines
+    # metadata:
+    #   annotations:
+    #     "compose.score.dev/publish-port": "9201"
+    #     "compose.score.dev/stack-version": "8.14.0"
```
[the remark this answers is not in the document]
Comment (giesan) [reply examples/14-elasticsearch/README.md:15]: Thanks

File: examples/14-elasticsearch/score.yaml (hunk 3 above)
Comment (giesan) [reply examples/14-elasticsearch/score.yaml:13]: The `ca.crt` certificate is located in the container in the `/usr/share/elasticsearch/config/certs/ca/` directory and must be copied from there manually. A note about this is displayed in the info block after the docker-compose.yaml file is created in the console.

```shell
INFO: elasticsearch.default#example.ecs: elasticsearch.default#example.ecs: To connect to elasticsearch:
 download certificate from container per command like:
        docker cp [CONTAINER-NAME]:/usr/share/elasticsearch/config/certs/ca/ca.crt /tmp/
 and than check connection per culr like:
        curl --cacert /tmp/ca.crt -u elastic:lj6LSY7XvqY4Ibfe https://localhost:9200
DEBUG: Provisioned resource 'elasticsearch.default#example.ecs'
```

I can't find a helpful approach for the example file.
Do you have a better suggestion than copying the certificate manually?

Comment (astromechza) [reply examples/14-elasticsearch/score.yaml:13]: Ok I see. Ok that's probably alright then. Unless we just turn TLS off completely

**Review by astromechza — APPROVED** [review]

File: examples/14-elasticsearch/score.yaml (hunk 3 above)
Comment (giesan) [reply examples/14-elasticsearch/score.yaml:13]: The [article](https://www.elastic.co/blog/getting-started-with-the-elastic-stack-and-docker-compose/) explicitly states that security should not be deactivated.

> **Setup and Elasticsearch node**
> 
> One of the first bits of trouble that's often run into when getting started is security configuration. As of 8.0, security is enabled by default. Therefore, we'll need to make sure we have the certificate CA setup correctly by utilizing a "setup" node to establish the certificates. Having [security enabled is a recommended practice](https://www.elastic.co/guide/en/elasticsearch/reference/current/secure-cluster.html) and should not be disabled, even in POC environments.

That's why I think we should leave it active.

**Pull request merged**